### DISCUSSION TITLE
Document variable usage

### DISCUSSION THREAD

**Discussion by balazsmaria** [graphql-java-kickstart/graphql-spring-webclient#20] [Q&A] [answered]
Looking at GraphQLRequestBuilder.variables(Object variables) it's not entirely clear how to set variables.

**Comment by balazsmaria** [comment, marked as answer]
What I did was:

```
		Map<String, String> vars = Map.of("myvar", "value");

		var request = GraphQLRequest.builder()
				.header(HttpHeaders.AUTHORIZATION, TOKEN)
				.resource("query.graphql")
				.variables(vars)
				.build();
```

**Reply by oliemansm** [reply]
You can indeed use that as you've figured out or you could pass in a custom object that'll be serialized by Jackson.

**Comment by D0mc3k** [comment]
I have used an immutable map 
Map<String, Object> variables = ImmutableMap.of(key, value);

**Comment by Jwqm** [comment]
A query, I have this query in graphql, 

query {
  parameter(
    input:{
       name: "HOLA"
    }
  )
  {
    name
    value
  }
}

How should my GraphQLRequest look?
